core/src/main.cpp

```diff
@@ -54,11 +56,11 @@ int main(int argc, char* argv[])
         // Apply the model configuration
         model.configure();
         // Run the Model
-        model.run();
+        return_code = model.run();
```

Comment:
Since the exceptions are not actually being handled at the lower levels, if returning a return code is necessary, rather than just letting the unhandled exception escape main, do it handle the exception here.